### PR TITLE
Fix/lp 1845696

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

Check the role binding has been deleted before starting to create a new one;

## QA steps

- deploy any caas charm has RBAC resources;
- upgrade charm to see if role binding could be created;

## Documentation changes

None

## Bug reference

https://bugs.launchpad.net/juju/+bug/1845696

### CODE REVIEW

**Comment by ycliuhw** [comment]
[https://bugs.launchpad.net/juju/+bug/1845696](https://trello.com/c/L0nSWKCV/927-https-bugslaunchpadnet-juju-bug-1845696)

**Comment by ycliuhw** [comment]
[Juju won't create service account if role binding already exists](https://trello.com/c/zNUZ9i4g/921-juju-wont-create-service-account-if-role-binding-already-exists)

File: caas/kubernetes/provider/rbac.go [hunk 1]
```diff
@@ -437,10 +439,39 @@ func (k *kubernetesClient) ensureRoleBinding(rb *rbacv1.RoleBinding) (out *rbacv
...
 	for _, v := range rbs {
 		if v.GetName() == rb.GetName() {
-			if err := k.deleteRoleBinding(v.GetName(), v.GetUID()); err != nil {
+			name := v.GetName()
+			UID := v.GetUID()
+
+			if err := k.deleteRoleBinding(name, UID); err != nil {
+				return nil, cleanups, errors.Trace(err)
+			}
+			notReadyYetErr := errors.New(fmt.Sprintf("role binding %q is still being deleted", name))
```
Comment (wallyworld) [inline caas/kubernetes/provider/rbac.go]: let's break this out to a separate helper function

**Comment by ycliuhw** [comment]
non related FAIL: uniter_test.go:940: UniterSuite.TestUniterRelations
!!build!!

**Comment by ycliuhw** [comment]
13:26:45 util_unix_test.go:118:
!!build!!

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**